tests/time_integration/bdf_time_derivative_01.cc

```diff
@@ -0,0 +1,119 @@
+ *  ______________________________________________________________________
+ */
+
+#include <iostream>
+#include <numeric>
+
+#include <deal.II/lac/la_vector.h>
+
+#include <exadg/time_integration/bdf_constants.h>
+
+// Check BDF time derivative given times and quantities
+
+using namespace ExaDG;
+using VectorType = dealii::LinearAlgebra::Vector<double>;
+
+void
+test_constant_quantites(unsigned int const order)
+{
+  std::cout << "Check derivative of constant quantitites" << std::endl;
+
+  // construct constant qunatities
+  std::vector<VectorType> constant_quantities_np_values;
+  for(unsigned int i = 0; i < order + 1; ++i)
+  {
+    VectorType vec(1);
+    vec[0] = 1.0;
+    constant_quantities_np_values.push_back(vec);
+  }
+
+  std::vector<VectorType const *> constant_quantities_np(constant_quantities_np_values.size());
+  std::transform(constant_quantities_np_values.begin(),
+                 constant_quantities_np_values.end(),
+                 constant_quantities_np.begin(),
+                 [](VectorType const & t) { return &t; });
+
+
+  // construct times
+  std::vector<double> times_np(order + 1);
+  std::iota(times_np.begin(), times_np.end(), 0.0);
+  std::sort(times_np.begin(), times_np.end(), [](auto const a, auto const b) { return a > b; });
+
+  // compute temporal derivative
+  VectorType derivative(1);
+  derivative[0] = -99.0;
+
+  compute_bdf_time_derivative(derivative, constant_quantities_np, times_np);
+
+  // check if derivative is 0
+  assert(derivative[0] < 1e-12);
+  std::cout << "OK" << std::endl;
+}
+
+void
+test_linear_changing_quantites(unsigned int const order)
+{
+  std::cout << "Check derivative of linear changing quantitites" << std::endl;
+
+  // construct qunatities with slope -1
+  std::vector<VectorType> constant_quantities_np_values;
+  for(unsigned int i = 0; i < order + 1; ++i)
+  {
+    VectorType vec(1);
+    vec[0] = i;
+    constant_quantities_np_values.push_back(vec);
+  }
+
+  std::vector<VectorType const *> constant_quantities_np(constant_quantities_np_values.size());
+  std::transform(constant_quantities_np_values.begin(),
+                 constant_quantities_np_values.end(),
+                 constant_quantities_np.begin(),
+                 [](VectorType const & t) { return &t; });
+
+
+  // construct times
+  std::vector<double> times_np(order + 1);
+  std::iota(times_np.begin(), times_np.end(), 0.0);
+  std::sort(times_np.begin(), times_np.end(), [](auto const a, auto const b) { return a > b; });
+
+
+  // compute temporal derivative
+  VectorType derivative(1);
+  derivative[0] = -99.0;
+
+  compute_bdf_time_derivative(derivative, constant_quantities_np, times_np);
+
+  // check if derivative is -1
+  assert(std::abs(derivative[0] + 1.0) < 1e-12);
+  std::cout << "OK" << std::endl;
+}
+
+int
+main()
+{
+  for(unsigned int order = 1; order <= 4; ++order)
+  {
+    std::cout << "Order " << order << std::endl;
+    test_constant_quantites(order);
```

Comment:
This tests in not more than checking that the sum of all bdf-constants is zero, as introduced in #474. Since the function call `compute_bdf_derivative()` is tested with the second test below, I question whether we need this test. I also do not like that the two functions copy so many lines of code. This implies unnecessary maintenance effort for the future.

Reply:
It uses the different interface which constructs the BDF object internally. I think we should test it. In my opinion, the more we test, the better.

It is a test and should not be touched, so maintenance will not be a problem. I will adapt it, but I really think we should keep both tests.

Reply:
> It is a test and should not be touched, so maintenance will not be a problem.

I don't think this is correct. I will remember having you as a volunteer to adjust tests for whatever reasons due to code changes in the future :). 
But seriously: A software needs continuous updates and refactoring. If one has to adjust 10 tests rather than 2, the motivation to make changes to a code will decrease a lot. I am convinced that one should not treat writing of tests shabbily for such reasons.